src/uu/runcon/src/runcon.rs

```diff
@@ -0,0 +1,422 @@
+// spell-checker:ignore (vars) RFILE
+
+use uucore::{executable, show_error, show_usage_error};
+
+use clap::{App, Arg};
+use selinux::{OpaqueSecurityContext, SecurityClass, SecurityContext};
+
+use std::borrow::Cow;
+use std::ffi::{CStr, CString, OsStr, OsString};
+use std::os::raw::c_char;
+use std::{io, ptr};
+
+mod errors;
+
+use errors::{report_full_error, Error, Result};
+
+static VERSION: &str = env!("CARGO_PKG_VERSION");
+static ABOUT: &str = "Run command with specified security context.";
+static DESCRIPTION: &str = "Run COMMAND with completely-specified CONTEXT, or with current or \
+                      transitioned security context modified by one or more of \
+                      LEVEL, ROLE, TYPE, and USER.\n\n\
+                      If none of --compute, --type, --user, --role or --range is specified, \
+                      then the first argument is used as the complete context.\n\n\
+                      Note that only carefully-chosen contexts are likely to successfully run.\n\n\
+                      With neither CONTEXT nor COMMAND are specified, \
+                      then this prints the current security context.";
+
+pub mod options {
+    pub static COMPUTE: &str = "compute";
+
+    pub static USER: &str = "user";
+    pub static ROLE: &str = "role";
+    pub static TYPE: &str = "type";
+    pub static RANGE: &str = "range";
+}
+
+fn get_usage() -> String {
+    format!(
+        "{0} [CONTEXT COMMAND [ARG...]]\n    \
+         {0} [-c] [-u USER] [-r ROLE] [-t TYPE] [-l RANGE] COMMAND [ARG...]",
+        executable!()
+    )
+}
+
+pub fn uumain(args: impl uucore::Args) -> i32 {
+    let usage = get_usage();
+
+    let config = uu_app().usage(usage.as_ref());
+
+    let options = match parse_command_line(config, args) {
+        Ok(r) => r,
+        Err(r) => {
+            if let Error::CommandLine(ref r) = r {
+                match r.kind {
+                    clap::ErrorKind::HelpDisplayed | clap::ErrorKind::VersionDisplayed => {
+                        println!("{}", r);
+                        return libc::EXIT_SUCCESS;
+                    }
+                    _ => {}
+                }
+            }
+
+            show_usage_error!("{}.\n", r);
+            return libc::EXIT_FAILURE;
+        }
+    };
+
+    match &options.mode {
+        CommandLineMode::Print => {
+            if let Err(r) = print_current_context() {
+                show_error!("{}.", report_full_error(&r));
+                return libc::EXIT_FAILURE;
```

Comment:
Could you change the exit code like this:
> Exit status:

> 126 if command is found but cannot be invoked
127 if runcon itself fails or if command cannot be found
the exit status of command otherwise

Reply:
Good catch! Thank you.

However, the GNU implementation does not behave like that. Here are all possible exit status of the GNU `runcon` when it is instructed to run a command (that is, not when printing):
- `127` if `execvp()` fails and `errno` is `ENOENT`.
- `126` if `execvp()` fails and `errno` is **not** `ENOENT`.
- `1` on all other errors.
- The exit status of the executed command.

Given that we strive for compatibility, I'm modifying our implementation to behave like so.